### ISSUE TITLE
Clarify situation about hibernate-validator + javax.el implementation dependencies

### ISSUE THREAD

**Issue by bclozel** [spring-projects/spring-boot#7500]
While working on the Spring Boot Web Reactive Starter, I ran into an issue. I recently added the `hibernate.validator` dependency to the `spring-boot-starter-web-reactive`, for feature parity with `spring-boot-starter-web`.
When starting an application with a Netty-based runtime, I'm getting the following:

```
Caused by: javax.validation.ValidationException: HV000183: Unable to load 'javax.el.ExpressionFactory'. Check that you have the EL dependencies on the classpath, or use ParameterMessageInterpolator instead
```

This is triggered by the `ValidationAutoConfiguration`. After chatting with @snicoll, a new guard Condition should be added in #6228 to prevent that auto-configuration from running if no el implementation is available.

This is obviously a known issue and I've seen the various discussions and how:
1) `spring-boot-starter-web` is relying on `hibernate-validator` and not `spring-boot-starter-validation` (see #5454)
2) `spring-boot-starter-validation` depends on `tomcat-embed-el`, see 346392d31b

While I understand those two points, I'm a bit stuck trying to find the best solution for a web reactive starter that does accept both servlet and non-servlet runtimes.

Is there something more we should discuss there or improved for supporting non-servlet runtimes?

**Comment by snicoll** [comment]
> After chatting with @snicoll, a new guard Condition should be added in #6228

On a second thought, it means that the auto-configuration has some Hibernate validator specific details which is something I'd like to avoid. We should probably attempt to get a validator in a custom condition.

**Comment by bclozel** [comment]
Discussed that with the team: creating a new netty-starter bringing its own EL implementation should take Netty on par with other choices of containers and solve this issue.

**Comment by danielfernandez** [comment]
That sounds good! Please note the exception also happened on Jetty and Undertow...

**Comment by bclozel** [comment]
Actually, this has been fixed by #7598 so no need to provide a default EL implementation. A `MessageInterpolatorFactory` will select the right dependency or use hibernate validator's `ParameterMessageInterpolator` as a fallback.
